### PR TITLE
Adding some form components in react-bootstrap

### PR DESCRIPTION
**Pull request by danilobjr**
Some components are missing in react-bootstrap.
So I added some of them like you see here: https://github.com/danilojrr/DefinitelyTyped/commit/a6a476190ce2197f04f541c360fdccc543f0005d

The new components was created based on react-bootstrap documentation. Topics: [Forms](https://react-bootstrap.github.io/components.html#forms) and [Supported Controls](https://react-bootstrap.github.io/components.html#forms-controls).

cc @walkerburgin @vsiao

### CODE REVIEW

**Comment by tht13** [comment]
Any progress on this? I would really like to use these features rather than write them out manually in standard html tags

**Comment by danilojrr** [comment]
@tht13 So do I. Waiting for the guys merge my PR.

File: react-bootstrap/react-bootstrap.d.ts [hunk 1]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
     }
     var FormControls: FormControlsClass;
 
+    // <Form />
+    // ----------------------------------------
+    interface FormProps extends React.HTMLAttributes {
+        bsClass?: string;
+        componentClass?: string;
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: I think this should be `React.ReactType`

File: react-bootstrap/react-bootstrap.d.ts [hunk 2]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
...
+        inline?: boolean;
+    }
+    type Form = React.ClassicComponent<FormProps, {}>;
+    var Form: React.ClassicComponentClass<FormProps>;
+
+    // <FormGroup />
+    // ----------------------------------------
+    interface FormGroupProps extends React.HTMLAttributes {
+        bsClass?: string;
+        bsSize?: 'sm' | 'small' | 'lg' | 'large';
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: nit: prefer double quotes `"sm"`

File: react-bootstrap/react-bootstrap.d.ts [hunk 3]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
...
+    // <Form />
+    // ----------------------------------------
+    interface FormProps extends React.HTMLAttributes {
+        bsClass?: string;
+        componentClass?: string;
+        horizontal?: boolean;
+        inline?: boolean;
+    }
+    type Form = React.ClassicComponent<FormProps, {}>;
+    var Form: React.ClassicComponentClass<FormProps>;
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: [`Form`](https://github.com/react-bootstrap/react-bootstrap/blob/master/src/Form.js) is a modern component, not classic

File: react-bootstrap/react-bootstrap.d.ts [hunk 4]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
...
+    // <FormGroup />
+    // ----------------------------------------
+    interface FormGroupProps extends React.HTMLAttributes {
+        bsClass?: string;
+        bsSize?: 'sm' | 'small' | 'lg' | 'large';
+        controlId?: string;
+        validationState?: 'success' | 'warning' | 'error';
+    }
+    type FormGroup = React.ClassicComponent<FormGroupProps, {}>;
+    var FormGroup: React.ClassicComponentClass<FormGroupProps>;
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: ditto, should be `class FormGroup extends React.Component<FormGroupProps, {}> {`

File: react-bootstrap/react-bootstrap.d.ts [hunk 5]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
...
+    
+    // <ControlLabel />
+    // ----------------------------------------
+    interface ControlLabelProps extends React.HTMLAttributes {
+        bsClass?: string;
+        htmlFor?: string;
+        srOnly?: boolean;
+    }
+    type ControlLabel = React.ClassicComponent<ControlLabelProps, {}>;
+    var ControlLabel: React.ClassicComponentClass<ControlLabelProps>;
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: modern

File: react-bootstrap/react-bootstrap.d.ts [hunk 6]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
...
+    
+    // <FormControl.Feedback />
+    // ----------------------------------------
+    interface FormControlFeedbackProps extends React.HTMLAttributes {
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: [docs] mention `bsClass?: string`?

File: react-bootstrap/react-bootstrap.d.ts [hunk 7]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
...
+    }
+    class FormControlFeedback extends React.Component<FormControlFeedbackProps, {}> {
+    }
+    
+    // <FormControl />
+    // ----------------------------------------
+    interface FormControlProps extends React.HTMLAttributes {
+        bsClass?: string;
+        componentClass?: string;
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: `React.ReactType`

File: react-bootstrap/react-bootstrap.d.ts [hunk 8]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
...
+        id?: string;
+        type?: string;
+    }
+    interface FormControlClass extends React.ClassicComponentClass<FormControlProps> {
+        Feedback: typeof FormControlFeedback;
+    }
+    type FormControl = React.ClassicComponent<FormControlProps, {}>;
+    var FormControl: FormControlClass;
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: modern

File: react-bootstrap/react-bootstrap.d.ts [hunk 9]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
...
+    
+    // <HelpBlock />
+    // ----------------------------------------
+    interface HelpBlockProps extends React.HTMLAttributes {
+        bsClass?: string;
+    }
+    type HelpBlock = React.ClassicComponent<HelpBlockProps, {}>;
+    var HelpBlock: React.ClassicComponentClass<HelpBlockProps>;
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: modern

File: react-bootstrap/react-bootstrap.d.ts [hunk 10]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
...
+    // <Checkbox />
+    // ----------------------------------------
+    interface CheckboxProps extends React.HTMLAttributes {
+        bsClass?: string;
+        disabled?: boolean;
+        inline?: boolean;
+        validationState?: 'success' | 'warning' | 'error';
+    }
+    type Checkbox = React.ClassicComponent<CheckboxProps, {}>;
+    var Checkbox: React.ClassicComponentClass<CheckboxProps>;
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: modern

File: react-bootstrap/react-bootstrap.d.ts [hunk 11]
```diff
@@ -704,6 +704,88 @@ declare module "react-bootstrap" {
...
+    // <Radio />
+    // ----------------------------------------
+    interface RadioProps extends React.HTMLAttributes {
+        bsClass?: string;
+        disabled?: boolean;
+        inline?: boolean;
+        validationState?: 'success' | 'warning' | 'error';
+    }
+    type Radio = React.ClassicComponent<RadioProps, {}>;
+    var Radio: React.ClassicComponentClass<RadioProps>;
```
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: modern

**Comment by vsiao** [comment]
Sorry for the delay. Let me know if you have any questions.

File: react-bootstrap/react-bootstrap.d.ts (hunk 1 above)
Comment (danilojrr) [inline react-bootstrap/react-bootstrap.d.ts]: Yeah. I would like to use ReactType, but in another interfaces (like ButtonProps) there's a `componentClass` property with type `string`. Others with `any` type. I just follow the "pattern". 
There's several TODOs comments in the file. We need to fix this.

File: react-bootstrap/react-bootstrap.d.ts (hunk 2 above)
Comment (danilojrr) [inline react-bootstrap/react-bootstrap.d.ts]: Ok. I'll put double quotes.

**Comment by danilojrr** [comment]
@vsiao Not a problem. =]
What I do when I fix all this stuff? Another PR? Or there's a way to put my new commit here?

**Comment by vsiao** [comment]
You can just amend your commit, push to your `react-bootstrap` branch and we'll see the changes here.

File: react-bootstrap/react-bootstrap.d.ts (hunk 1 above)
Comment (vsiao) [inline react-bootstrap/react-bootstrap.d.ts]: Yeah, those should probably be `ReactType` too.

File: react-bootstrap/react-bootstrap.d.ts (hunk 6 above)
Comment (danilojrr) [inline react-bootstrap/react-bootstrap.d.ts]: Yes. Here: [https://react-bootstrap.github.io/components.html#forms-props-form-control](https://react-bootstrap.github.io/components.html#forms-props-form-control)

**Comment by danilojrr** [comment]
So, I changed classic components to modern components and `componentClass` properties now have `React.ReactType` as its type. Check it out if I did correctly.

**Comment by tht13** [comment]
Is this ready to merge now?

**Comment by vsiao** [comment]
lgtm. @vvakame?

**Comment by vsiao** [comment]
@danilojrr since this PR fixes #9062, can you mention that in your commit message so that the issue will be closed when this PR is merged? thanks.

**Comment by danilojrr** [comment]
@vsiao I cant change a commit message. How can I do this?

**Comment by vsiao** [comment]
`git rebase -i HEAD^^` to squash your commits or `git commit --amend` to amend your previous comit message

**Comment by danilojrr** [comment]
@vsiao It's done. I think this is ready for merge. And thank you for tips on git stuff.

**Pull request merged**